src/plugins/colum-chooser/column-chooser.component.ts

```diff
@@ -0,0 +1,231 @@
+import {Component, Optional, OnInit, OnDestroy, Input, Output, EventEmitter, ChangeDetectorRef} from '@angular/core';
+import {Command} from 'ng2-qgrid/core/command';
+import {PipeUnit} from 'ng2-qgrid/core/pipe/pipe.unit2';
+import {GridService} from 'ng2-qgrid/main/grid';
+import * as columnService from 'ng2-qgrid/core/column/column.service';
+import {isFunction, noop} from 'ng2-qgrid/core/utility';
+import {Aggregation} from 'ng2-qgrid/core/services';
+
+import {RootService} from 'ng2-qgrid/infrastructure/component';
+import {PluginComponent} from '../plugin.component';
+import {TemplatePath} from 'ng2-qgrid/core/template';
+
+const GRID = 'qGrid';
+const COLUMN_CHOOSER_NAME = `${GRID}ColumnChooser`;
+
+@Component({
+	selector: 'q-grid-column-chooser',
+	template: `
+		<ng-container key="column-chooser.tpl" [context]="context"></ng-container>
+	`
+	// templateUrl: './column-chooser.tpl.html'
+})
+export class ColumnChooserComponent extends PluginComponent implements OnInit, OnDestroy {
+	@Input('canAggregate') columnChooserCanAggregate: boolean;
+	@Output('submit') submitEvent = new EventEmitter<any>();
+	@Output('cancel') cancelEvent = new EventEmitter<any>();
+
+	private service: any;
+	private aggregations: any;
+	private origin: any;
+	private filter: string = '';
+
+	private columns: any[];
+
+	constructor(@Optional() root: RootService, private gridService: GridService, private cdRef: ChangeDetectorRef) {
+		super(root);
+
+		this.models = ['columnChooser'];
+		this.initColumns();
+		this.defaultSelected = true;
+	}
+
+	private get columSelectors(): any[] {
+		return this.columns.filter(column => {
+			return column.title && column.key.includes(this.filter);
+		});
+	}
+
+	private get allSelected(): boolean {
+		return this.stateAll();
+	}
+	private set allSelected(value: boolean) {
+		for (const column of this.columSelectors) {
+			column.isVisible = value;
+		}
+		this.service.invalidate('column.chooser', {}, PipeUnit.column);
+	}
+
+	private get defaultSelected(): boolean {
+		return this.columSelectors.every(c => (c.isDefault !== false && c.isVisible !== false) ||
+			(c.isDefault === false && c.isVisible === false));
+	}
+	private set defaultSelected(value: boolean) {
+		if (value) {
+			for (const column of this.columSelectors) {
+				column.isVisible = column.isDefault !== false;
+			}
+			this.service.invalidate('column.chooser', {}, PipeUnit.column);
+		}
+	}
+
+	private get allSelectedIndeterminate(): boolean {
+		return !this.allSelected &&
+			this.columSelectors.some(this.columnState.bind(this));
+	}
+
+	private stateAll() {
+		const allSet = this.columSelectors.every(this.columnState.bind(this));
+		return allSet;
+	}
+
+	private columnState(column, value) {
+		return column.isVisible;
+	}
+
+	private setColumnState(column, value) {
+		column.isVisible = value;
+		this.service.invalidate('column.chooser', {}, PipeUnit.column);
+	}
+
+	private isRowOptions(column) {
+		return column.key === 'rowOptions';
```

Comment:
wrong should be filtered by class, please reuse ng1 code, if there are bugs please let me know

Reply:
fixed (the method removed, as is not used)